src/firefly/java/edu/caltech/ipac/visualize/plot/FitsRead.java

```diff
@@ -69,7 +69,7 @@
     private final BasicHDU hdu;
     private float[] float1d;
     private Fits fits;
-    private ImageHeader imageHeader;
+    private static ImageHeader imageHeader;
```

Comment:
Will it still be thread-safe after it's changed to static?

Reply:
Good point.

Reply:
The ImageHeader is an instance variable and should not changed to static.  Remove the static, north-up, return to default, rotation all work fine.  The only issue is the wcs match when the image is MSX.  This issue was fixed in the dev.  But it is an existing issue in gwt_rc.